### PR TITLE
jbranchaud/er 54 create module completion certificate on module finish

### PR DESCRIPTION
**Pull request by jbranchaud**
I reworked the pnpm-lock.json changes to deal with an incoming merge conflict from `main`. Instead of force pushing over @zacjones93's branch, I opened a separate PR. If we feel good about this one, we can merge this and close the other.

- **feat: download module certificate on [module]/completed**
- **update workshop-template with correct module type**
- **fix type error for confetti as any**
- **Add complete and continue button**
- **Add lockfile changes**

merge

### CODE REVIEW

**Review by DrShpongle — APPROVED** [review]
Thanks Josh!

File: apps/epic-react/src/certificate/certificate-form.tsx [hunk 1]
```diff
@@ -53,45 +53,51 @@ const CertificateForm: React.FC<React.PropsWithChildren<{module: Module}>> = ({
...
-      <Dialog.Overlay className="fixed inset-0 z-30 bg-black/20 backdrop-blur-sm" />
+      <Dialog.Overlay className="bg-black/20 fixed inset-0 z-30 backdrop-blur-sm" />
       <Dialog.Content className="fixed left-[50%] top-[50%] z-50 max-h-[85vh] w-[90vw] max-w-[450px] translate-x-[-50%] translate-y-[-50%] rounded-[6px] border border-gray-100 bg-white p-[25px] shadow-2xl shadow-black/20 focus:outline-none dark:border-gray-700/50 dark:bg-gray-800 dark:shadow-black/40">
         <Dialog.Title className=" pb-3 text-xl font-semibold dark:border-gray-700">
           Your Certificate of Completion
         </Dialog.Title>
         <div className="pointer-events-none relative flex items-center justify-center">
           <Image
-            src="https://res.cloudinary.com/dg3gyk0gu/image/upload/v1700132958/skill-template-certificate-thumbnail.png"
+            src="/assets/epic-react-certificate-bg.svg"
```
Comment (zacjones93) [inline apps/epic-react/src/certificate/certificate-form.tsx:79]: update certificate with epic react template. @DrShpongle the image generated is a little fuzzy, not sure if there are easy optimizations there

File: apps/epic-react/src/certificate/certificate-form.tsx [hunk 2]
```diff
@@ -144,7 +150,7 @@ const CertificateForm: React.FC<React.PropsWithChildren<{module: Module}>> = ({
                 )}
               >
                 <a
-                  download
+                  download={`${module.slug.current}-certificate.png`}
```
Comment (zacjones93) [inline apps/epic-react/src/certificate/certificate-form.tsx:153]: this sets the file name for the certificate that is downloaded

File: apps/epic-react/src/components/lesson-completion-toggle.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  React.useEffect(() => {
+    if (toggleClicked && moduleCompleted) {
+      router.push(`/modules/${module.slug.current}/completed`)
+    } else if (toggleClicked && optimisticallyToggled) {
+      reward()
+      router.push(
+        `/modules/${module.slug.current}/${moduleProgress?.nextLesson?.slug}`,
+      )
+    }
+  }, [toggleClicked, moduleProgress, optimisticallyToggled])
```
Comment (zacjones93) [inline apps/epic-react/src/components/lesson-completion-toggle.tsx:52]: watching if the completion button is clicked, if it is we check if the module is completed. If the module is completed we navigate to the `[module]/completed` route. If it isn't then we navigate to the next lesson.

File: apps/epic-react/src/components/lesson-completion-toggle.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import {useReward} from 'react-rewards'
+import {useLesson} from '@skillrecordings/skill-lesson/hooks/use-lesson'
+import {useRouter} from 'next/router'
+import {trpcSkillLessons} from '@skillrecordings/skill-lesson/utils/trpc-skill-lessons'
+import toast from 'react-hot-toast'
+
+const LessonCompleteToggle = () => {
+  const {module} = useLesson()
+  const router = useRouter()
+  const {reward, isAnimating} = useReward('rewardId', 'confetti')
```
Comment (zacjones93) [inline apps/epic-react/src/components/lesson-completion-toggle.tsx:12]: react-rewards is hooks-based now which is much easier than the epic-react-gatsby set up

File: apps/epic-react/src/pages/modules/[module]/completed.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  })
+
+  if (!moduleCompleted) {
+    return {
+      redirect: {
+        destination: `/modules/${module?.slug.current}/${nextLesson?.slug}`,
+        permanent: false,
+      },
+    }
+  }
```
Comment (zacjones93) [inline apps/epic-react/src/pages/modules/[module]/completed.tsx:55]: if module isn't completed they can't access this page

File: apps/epic-react/src/pages/modules/[module]/completed.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                  {/* </g> */}
+                </motion.svg>
+                Great work!
+              </h1>
+              {module && (
+                <p className="mb-8 text-center text-base  text-gray-600 sm:text-lg md:text-left">
+                  You&apos;ve completed {module.title} workshop.
+                </p>
+              )}
+              {module && <ModuleCertificate module={module} />}
```
Comment (zacjones93) [inline apps/epic-react/src/pages/modules/[module]/completed.tsx:138]: using current skill implementation of module certificate as the logic was already in here vs porting over the epic-react-gatsby implementation

File: packages/skill-lesson/lib/modules.ts [hunk 7]
```diff
@@ -184,7 +184,7 @@ export const getModuleById = async (id: string) =>
         _type,
         title,
         "slug": slug.current,
-        "image": image.asset->url,
+        "image": image.url,
```
Comment (zacjones93) [inline packages/skill-lesson/lib/modules.ts:187]: for epic react, this was the structure needed to get the module image. Making a change to a monorepo package makes me a little nervous though

**Pull request merged**